### PR TITLE
[#8] Add interfaces for Maps

### LINKED ISSUE #8: Add type class for maps

### PR DESCRIPTION
**Pull request by ChShersh**
Resolves #8 

What's done:

* `Universum.Container.Class` renamed to `Universum.Container.One`
* `ToPairs` class removed. Now we have functions that work for every `IsList`
* `StaticMap` and `DynamicMap` type classes introduced
* Instances for `StaticMap` are implemented
* Instances for `DynamicMap` are implemented

### CODE REVIEW

File: src/Universum/Container/Map.hs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    member = IS.member
+    lookup k m = guard (member k m) *> pure k
+    {-# INLINE size #-}
+    {-# INLINE lookup #-}
+    {-# INLINE member #-}
+
+-- | Operator version of 'lookup' function.
+infixl 9 !?
+(!?) :: StaticMap t => t -> Key t -> Maybe (Val t)
+(!?) m k = lookup k m
```
Comment (vrom911) [inline src/Universum/Container/Map.hs]: 😍

File: universum.cabal [hunk 2]
```diff
@@ -82,7 +83,6 @@ library
                      , transformers
                      , unordered-containers
                      , utf8-string
-                     , vector
```
Comment (vrom911) [inline universum.cabal]: 👍

File: src/Universum/Container/Map.hs [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+{- | Read-only map or set. Contains polymorhic functions which work for both
+sets and maps.
+-}
+class StaticMap t where
+    type Key t :: Type
+    type Val t :: Type
+
+    size   :: t -> Int
+    lookup :: Key t -> t -> Maybe (Val t)
+    member :: Key t -> t -> Bool
```
Comment (vrom911) [inline src/Universum/Container/Map.hs]: How do you think, will it make sense to add `lookupDefault` as well?

**Comment by ChShersh** [comment]
@vrom911 You can review again!

**Pull request merged**